filecoin-proofs/src/api/seal.rs

```diff
@@ -537,6 +543,346 @@ pub fn seal_commit_phase2<Tree: 'static + MerkleTreeTrait>(
+        ensure!(comm_r != [0; 32], "Invalid all zero commitment (comm_r)");
+
+        let replica_id = generate_replica_id::<Tree::Hasher, _>(
+            &prover_id,
+            sector_id.into(),
+            &ticket,
+            comm_d,
+            &porep_config.porep_id,
+        );
+
+        let comm_r_safe = as_safe_commitment(&comm_r, "comm_r")?;
+        let comm_d_safe = DefaultPieceDomain::try_from_bytes(&comm_d)?;
+
+        let public_inputs = stacked::PublicInputs {
+            replica_id,
+            tau: Some(stacked::Tau {
+                comm_d: comm_d_safe,
+                comm_r: comm_r_safe,
+            }),
+            k: None,
+            seed,
+        };
+
+        let compound_setup_params = compound_proof::SetupParams {
+            vanilla_params: setup_params(
+                PaddedBytesAmount::from(porep_config),
+                usize::from(PoRepProofPartitions::from(porep_config)),
+                porep_config.porep_id,
+                porep_config.api_version,
+            )?,
+            partitions: Some(usize::from(PoRepProofPartitions::from(porep_config))),
+            priority: false,
+        };
+
+        let compound_public_params = <StackedCompound<Tree, DefaultPieceHasher> as CompoundProof<
+                StackedDrg<'_, Tree, DefaultPieceHasher>,
+            _,
+            >>::setup(&compound_setup_params)?;
+
+        let partitions = <StackedCompound<Tree, DefaultPieceHasher> as CompoundProof<
+                StackedDrg<'_, Tree, DefaultPieceHasher>,
+            _,
+            >>::partition_count(&compound_public_params);
+
+        // These are returned for aggregated proof verification.
+        let cur_inputs: Vec<_> = (0..partitions)
+            .into_par_iter()
+            .map(|k| {
+                StackedCompound::<Tree, DefaultPieceHasher>::generate_public_inputs(
+                    &public_inputs,
+                    &compound_public_params.vanilla_params,
+                    Some(k),
+                )
+            })
+            .collect::<Result<_>>()?;
+
+        inputs.extend(cur_inputs.clone());
+        inputs_added += 1;
+
+        // If we're at the last one and not at the pow2 target,
+        // duplicate it until we are.
+        if i == comm_rs.len() - 1 {
+            while inputs_added != target {
+                inputs.extend(cur_inputs.clone());
+                inputs_added += 1;
+            }
+        }
+    }
+
+    Ok(inputs)
+}
+*/
+
+pub fn aggregate_seal_commit_proofs<Tree: 'static + MerkleTreeTrait>(
+    porep_config: PoRepConfig,
+    commit_outputs: &[SealCommitOutput],
+) -> Result<AggregateSnarkProof> {
+    info!("aggregate_seal_commit_proofs:start");
+
+    let partitions = usize::from(PoRepProofPartitions::from(porep_config));
+    let verifying_key = get_stacked_verifying_key::<Tree>(porep_config)?;
+    let mut proofs: Vec<_> = commit_outputs
+        .iter()
+        .fold(Vec::new(), |mut acc, commit_output| {
```

Comment:
you can use [`flat_map`](https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.flat_map), makes it a little easier to read